index.js

```diff
@@ -0,0 +1,145 @@
+const baseUrl = 'https://trektravel.herokuapp.com/trips'
+
+// Get the modal
+const modal = document.getElementById('myModal');
+
+// Get the button that opens the modal
+const btn = document.getElementById("myBtn");
+
+// Get the <span> element that closes the modal
+const span = document.getElementsByClassName("close")[0];
+
+
+const successCallback = function(response){
+  console.log("POST request to reserve a spot on a trip was successful");
+  console.log(response);
+  let name = $('#trip-form').data('name');
+  $('#message').html(`Congratulations! You successfully reserved a spot on ${name}!`);
+
+  // When the user clicks the button, open the modal
+  modal.style.display = "block";
+};
+
+const continents = {
+  'Africa': 'africa.png',
+  'Antarctica': 'antarctica.png',
+  'Asia': 'asia.png',
+  'Australasia': 'australia.png',
+  'Europe': 'europe.png',
+  'North America': 'north_america.png',
+  'South America': 'south_america.png'
+};
+
+const continentImgInfo = function(trip){
+  if (continents[trip]){
+    return `<img src="img/${continents[trip]}" alt="${trip} picture" class="continent"/>`;
+  }
+};
+
+$(document).ready(()=> {
+  $('#reserve-form').hide();
+  $('#content').hide();
+  $('#show-trip').hide();
+
+  let loadTrips = function loadTrips() {
+    $.get(baseUrl, (response) => {
+      console.log(response);
+      console.log('success!');
+      response.forEach(function(trip) {
+        let a = continentImgInfo(trip.continent);
+        let tripInfo = `<li><h3 data-id=${trip.id} data-name='${trip.name}'>` + trip.name + ` | <span>${a}</span> | ${trip.weeks}<h3></li><hr>`;
+        $('#show-trips ul').append(tripInfo);
+      });
+    })
+    .fail(function(response){
+      console.log(response);
+      alert('Request was unsuccessful');
+    });
+  };
+
+  let loadTrip = function loadTrip(id){
+    $.get(baseUrl + `/${id}`, (response) => {
+      console.log('success!');
+      console.log(response);
+
+      let tripInfo = `
+      <p><span class='info'>ID:</span> ${response.id}<br>
+      <span>Category:</span> ${response.category}<br>
+      <span>Destination:</span> ${response.continent}<br>
+      <span>Details:</span> ${response.about}<br>
+      <span>Cost:</span> $${response.cost}<br>
+      <span>Length:</span> ${response.weeks} weeks</p>`;
+      $('#hook').html(tripInfo);
+
+      let button = $(`<button id="reserve" class="natGeoButton" value="${response.id}">Sign up today!</button>
+      `);
+
+      button.click((event) => {
+        $('#reserve-form').show();
```

Comment:
This is a callback inside a callback inside a callback! For me, 3 levels of nesting is usually an indicator that it's time to break some of these (probably the AJAX success callback) out into a separate named function.

See also: [callback hell](http://callbackhell.com/).